### DISCUSSION TITLE
How to add the link for additional files, that are not supposed to be extracted / expanded.. ??

### DISCUSSION THREAD

**Discussion by anderlli0053** [ScoopInstaller/Scoop#5198] [Help and Questions] [answered]
Is there a way to just download additional files, without extracting them?

Because Scoop tries to extract everything that is included into JSON URL array... even the files that are not supposed to be touched / extracted..

```
"url": [
        "https://www.somelink.com/files_to_be_extracted_shimmed_and_shortcutted.zip",
        "https://www.somelink.com/additional_files_not_to_be_touched.sdt"
    ]
```

**Comment by anderlli0053** [comment]
The only option that i can think of is using either `Invoke-RestMethod` or `Invoke-WebRequest` or even `Start-BitsTransfer` inside either `pre_install` or `post_install`... But does the Scoop itself has same kind of method?  I was looking here at the [Wiki](https://scoop-docs.vercel.app/docs/concepts/App-Manifests.html) but cannot find such command.. so i'm guessing that the command like that doesn't exists in Scoop.. ??

**Comment by rashil2000** [comment, marked as answer]
Scoop extracts only files which have archive extensions. You can rename the downloaded file by appending a non-archive extension, i.e. `url#/dl.zip_`

**Reply by anderlli0053** [reply]
> Scoop extracts only files which have archive extensions. You can rename the downloaded file by appending a non-archive extension, i.e. `url#/dl.zip_`

Ok, but in my case i have a custom binary file and therefore custom file extension `.SDT` , and it still tries to force extract it..  Do i really need to use URL fragment?

**Reply by anderlli0053** [reply]
I just used Combination of your suggested URL fragment + `Rename-Item \"$dir\\ .......` PowerShell command and now it works great. Thanks. The problem was that the Scoop forcefully tried to extract the files that weren't even archives, neither by file extension, nor content.

**Comment by niheaven** [comment]
It's strange that *.sdt shouldn't be extracted by any of extractor. What's your log?

**Reply by anderlli0053** [reply]
It is not an archive, it is custom binary file format and Scoop treated it like it is an actual archive of some kind, i resolved it by using url fragment + rename-item powershell command in post_install.

I changed the Scoop Config branch to "develop" and updated it, now it works fine.  I remember that, while Scoop extracted my binary, it extracted it in a way .EXE would appear (.bss segment, .data, ....). All of those segments out..... However, for now it is resolved, i haven't checked any logs before as i resolved it manually somehow. If it appears again i'll post the log here

**Reply by niheaven** [reply]
Alright, if you meet the bug in 'normal' setting (not renamed), please report again.

**Reply by anderlli0053** [reply]
> Alright, if you meet the bug in 'normal' setting (not renamed), please report again.

Will do, thanks
